data-processing-lib/ray/src/data_processing_ray/runtime/ray/ray_utils.py

```diff
@@ -109,16 +111,30 @@ def operator() -> ActorHandle:
             time.sleep(creation_delay)
             return clazz.options(**actor_options).remote(params)
 
+        logger = get_logger(__name__)
         cls_name = clazz.__class__.__name__.replace('ActorClass(', '').replace(')','')
+        current = list_actors(filters=[("class_name", "=", cls_name), ("state", "=", "ALIVE")], limit=MAX_LIST)
+        c_len = len(current)
         actors = [operator() for _ in range(n_actors)]
-        for i in range(120):
-            time.sleep(1)
-            alive = list_actors(filters=[("class_name", "=", cls_name), ("state", "=", "ALIVE")])
-            if len(actors) == len(alive):
-                return actors
-        # failed - raise an exception
-        print(f"created {actors}, alive {alive}")
-        raise UnrecoverableException(f"out of {len(actors)} created actors only {len(alive)} alive")
+        overall = c_len + n_actors
+        if overall < MAX_LIST:
+            n_list = min(overall + 10, MAX_LIST)
+            for i in range(120):
+                time.sleep(1)
+                alive = list_actors(filters=[("class_name", "=", cls_name), ("state", "=", "ALIVE")], limit=n_list)
+                if len(alive) >= n_actors + c_len:
+                    return actors
+            # failed
```

Comment:
@blublinsky : I would suggest that we remove the code in lines 128-131, and we replace it with raising an `UnrecoverableException` for two reasons:
1. It is better to just let the user know if a ray cluster could not be allocated that matches the desired specifications, rather than run with less resources. 
2. I do not see a proper way to implement this solution.  The current code is broken, it will not run, because you cannot just subtract two lists using `-` and it is returning `list[ActorStatus]` instead of `list[ActorHandle]`.  It would be possible, maybe, to run something like this:
```
               alive_actor_ids = [a.actor_id for a in alive]
               current_actor_ids = [a.actor_id for a in current]
               return [
                   actor for actor in actors
                   if actor._actor_id in alive_actor_ids and actor._ray_actor_id not in current_actor_ids
               ]
```
But this is not a proper solution, as it is accessing a protected variable of the `ActorHandle` class.